x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpLogic } from '../../../shared/http';
+
+import { IndexingStatus } from '../schema/types';
+import { EngineDetails } from './types';
+
+interface EngineValues {
+  dataLoading: boolean;
+  engine: EngineDetails | {};
+  engineName: string;
+  isMetaEngine: boolean;
+  isSampleEngine: boolean;
+  hasSchemaConflicts: boolean;
+  hasUnconfirmedSchemaFields: boolean;
+  engineNotFound: boolean;
+}
+
+interface EngineActions {
+  setEngineData(engine: EngineDetails): { engine: EngineDetails };
+  setEngineName(engineName: string): { engineName: string };
+  setIndexingStatus(activeReindexJob: IndexingStatus): IndexingStatus;
+  setEngineNotFound(notFound: boolean): boolean;
+  clearEngine(): void;
+  initializeEngine(): void;
+}
+
+export const EngineLogic = kea<MakeLogicType<EngineValues, EngineActions>>({
+  path: ['enterprise_search', 'app_search', 'engine_logic'],
+  actions: {
+    setEngineData: (engine) => ({ engine }),
+    setEngineName: (engineName) => ({ engineName }),
+    setIndexingStatus: (activeReindexJob) => activeReindexJob,
+    setEngineNotFound: (notFound) => notFound,
+    clearEngine: () => ({}),
```

Comment:
This is not a blocking comment. When I see things like this in reducers I've been changed them to
```
clearEngine: true
```

Based on the [recommendation](https://kea.js.org/docs/guide/concepts#actions) from Kea.

> There's one shorthand that can be useful. In case your actions take no arguments (e.g. loadUsers), just pass true, or anything else that's not a function, instead of an arguments-to-payload serializer.

I've also been tweaking stuff like:
```
setEngineNotFound: (notFound) => notFound
```

To:
```
setEngineNotFound: (notFound) => ({ notFound })
```

Again, just based on Kea's recommendations:
> One more thing. It's strongly recommended to always return an object as a payload from your actions

I don't think it's worth changing, I'm just noting this here.

Reply:
These are super small changes that are a win for consistency & are easily verifiable via unit tests, so I'm more than happy to make those here!

Reply:
https://github.com/elastic/kibana/pull/82992/commits/d296e1aa94f3a2efe50cc09ffb76b76bac1148ab